src/components/instruction/associated-token/types.ts

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable @typescript-eslint/no-redeclare */
+
+import { enums, type, Infer } from "superstruct";
+import { PublicKeyFromString } from "validators/pubkey";
+
+export type CreateIdempotentInfo = Infer<typeof CreateIdempotentInfo>;
+export const CreateIdempotentInfo = type({
```

Comment:
Is it possible, instead, to _just_ export the const, and then use `Infer<typeof CreateIdempotentInfo>` wherever the _type_ is needed? This redeclaration is confusing.

Reply:
I copied it from another type.ts which is in this repo haha. I think we can have another refactor PR for all of those.